packages/rol/adapters/thyra/src/function/ROL_ThyraProductME_Constraint_SimOpt.hpp

```diff
@@ -89,11 +89,9 @@ class ThyraProductME_Constraint_SimOpt : public Constraint_SimOpt<Real> {
   }
 
   void value(Vector<Real> &c, const Vector<Real> &u, const Vector<Real> &z, Real &tol) {
-#ifdef  HAVE_ROL_DEBUG
-    //u and z should be updated in the update functions before calling value
-    TEUCHOS_ASSERT(!u_hasChanged(u));
-    TEUCHOS_ASSERT(!z_hasChanged(z));
-#endif
+
```

Comment:
I thought more about this, and I would prefer to keep the check only in debug, as before, and update where needed in ROL. Do you mind trying to do that? Hopefully you don't have to do that in many places. Same thing in the other functions/classes. The policy is: "if a function changes u or z, then it should call the update functions".